scripts/mds/c4.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import random
+from argparse import ArgumentParser, Namespace
+from glob import glob
+from itertools import islice
+from typing import Any, Dict, Iterable, List, Tuple
+
+import datasets
+from datasets import IterableDataset
+
+from composer.datasets.streaming import StreamingDatasetWriter
+
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments."""
+    args = ArgumentParser()
+    args.add_argument('--out_root', type=str, required=True)
+    args.add_argument('--shard_size_limit', type=int, default=1 << 27)
```

Comment:
How many shards does this work out to, out of curiosity?

Reply:
Update: testing on 4398 imagenet train shards, I got about a 6% slowdown in shard open/close time by breaking shards into directories of 100 or 1000 and about a 6% speedup by moving the entire flat directory to root level, which I guess is what we deserve for having shards stored as separate files